services/blockchain-indexer/shared/dataService/business/pos/stakers.js

```diff
@@ -93,36 +118,50 @@ const getStakers = async params => {
 		},
 		['stakerAddress', 'amount'],
 	);
-	if (resultSet.length) stakers.data.stakers = resultSet;
 
-	stakers.data.stakers = await BluebirdPromise.map(
-		stakers.data.stakers,
+	// Populate stakers name and prepare response
+	stakersResponse.data.stakers = await BluebirdPromise.map(
+		stakes,
 		async stake => {
-			const { name = null } = await getIndexedAccountInfo({ address: stake.stakerAddress, limit: 1 }, ['name']);
+			// Try to use cached staker name. Query DB if not found.
+			let stakerName = allowedStakersAddressNameMap[stake.stakerAddress];
+			if (!stakerName) {
+				const accountInfo = await getIndexedAccountInfo({
+					address: stake.stakerAddress,
+					limit: 1,
+				},
+				['name'],
+				);
+				stakerName = accountInfo.name;
+			}
```

Comment:
This piece of code will never be executed, because the response is filtered based on `stakerAddressQueryFilter` in L112.

Reply:
The filter/cache will only be used when `search` param is passed. otherwise, the cache will be empty.
ref: https://github.com/LiskHQ/lisk-service/blob/f75c1971da84b39ec439815522c0823c30b71f47/services/blockchain-indexer/shared/dataService/business/pos/stakers.js#L89-L107